public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useCallback, useEffect, useMemo } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { selectors as e2eSelectors } from '@grafana/e2e-selectors';
+import { config, featureEnabled } from '@grafana/runtime';
+import { SceneComponentProps, SceneObjectBase, SceneObjectRef, SceneObjectState } from '@grafana/scenes';
+import { Button, ClipboardButton, Divider, Spinner, Stack } from '@grafana/ui';
+import { contextSrv } from 'app/core/core';
+import {
+  useDeletePublicDashboardMutation,
+  useGetPublicDashboardQuery,
+  useUpdatePublicDashboardMutation,
+} from 'app/features/dashboard/api/publicDashboardApi';
+import { NoUpsertPermissionsAlert } from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/ModalAlerts/NoUpsertPermissionsAlert';
+import { Loader } from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboard';
+import {
+  generatePublicDashboardUrl,
+  PublicDashboard,
+  PublicDashboardShareType,
+} from 'app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils';
+import { DashboardScene } from 'app/features/dashboard-scene/scene/DashboardScene';
+import { DashboardInteractions } from 'app/features/dashboard-scene/utils/interactions';
+import { AccessControlAction } from 'app/types';
+
+import { EmailSharing } from './EmailShare/EmailSharing';
+import { PublicSharing } from './PublicShare/PublicSharing';
+import ShareAlerts from './ShareAlerts';
+import ShareTypeSelect from './ShareTypeSelect';
+
+export interface ShareExternallyDrawerState extends SceneObjectState {
+  shareType: SelectableValue<PublicDashboardShareType>;
+  options: Array<SelectableValue<PublicDashboardShareType>>;
+  dashboardRef: SceneObjectRef<DashboardScene>;
```

Comment:
Don't think you need to pass the ref since this is a scene object. You can access the dashboard using scene utils (`getDashboardSceneFor`). The case when passing ref may be beneficial is tests mainly as you don't need to prep the entire structure to get the dashboard scene in place.